WSUCptSCapstone-Fall2022Spring2023/psd-gamifiedapp
# Sprint5/sound effects

closes #55 

Demo Video Link: https://youtu.be/vRoy2UP0qNE

Overview:

This PR covers the addition of sound effects to the game. The main sound effects added were: button press, button hover, transition, success, failure, level complete, and text typing.

Details:

Several sound effects were added to the game under the directory Assets/Audio/SoundEffects

In addition, changes were made to a number of prefabs, and to several code files to play sounds at the appropriate times. Below are the main code files changed in this PR.

LevelPlayerController.cs: Changed to play end-of-level sound if all of the phases have been complete.

SequentialTyper.cs: Changed to play custom typing sound if one is assigned, and also given a slight delay on printing spaces to make printing behavior more visually and audibly satisfying.

UILayoutHandler.cs: Modified to play scene transition sound.


Testing:

The following testing procedure was used to test the new content.

1. Start Game
2. Confirm hover sounds on main menu
3. Navigate to level select
4. Check sounds for buttons in level select and manual
5. Enter Level 1
6. Confirm typing sounds in output window align with printing.
7. Check sounds for buttons in level player
8. Confirm failure sound
9. Confirm Success sound
10. Confirm sounds in escape menu and exit level
11. Select Challenge Level 1
12. Complete phase to confirm level-complete sound


In the current build all of the new and edited content passes these tests as expected.
